torch/autograd/function.py

```diff
@@ -7,16 +7,21 @@
 import warnings
 from collections import OrderedDict
 from typing import Any, List, Optional
+import torch.autograd.forward_ad as fwAD
 
 # Formerly known as: _ContextMethodMixin
 class FunctionCtx(object):
 
     def save_for_backward(self, *tensors: torch.Tensor):
         r"""Saves given tensors for a future call to :func:`~Function.backward`.
 
-        **This should be called at most once, and only from inside the**
-        :func:`forward` **method. This should only be called with input or
-        output tensors**
+        ``save_for_backward`` should be called at most once, only from inside the
+        :func:`forward` method, and only with tensors.
+
+        All tensors intended to be used in the backward pass should be saved
+        with ``save_for_backward`` (as opposed to directly on ``ctx``) to prevent
+        incorrect gradients and memory leaks, and enable the application of saved
```

Comment:
nit: remove one "and"

Reply:
Hmm I was thinking prevent applies to incorrect gradients and memory leaks only, but perhaps I can reword this better?

Reply:
Maybe On the one hand to prevent incorrect gradients and memory leak, on the other hand etc  ?

Reply:
Might be too verbose actually, yeah its a bit tricky to parse in its current state but I don't see an option that is uniformly better - will just keep for now.